partner_firstname/views/base_config_view.xml

```diff
@@ -0,0 +1,32 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- © 2015 Antiun Ingenieria S.L. - Antonio Espinosa
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html). -->
+<openerp>
+<data>
+
+<record id="view_general_configuration" model="ir.ui.view">
+    <field name="name">Add partner_names_order config parameter</field>
+    <field name="model">base.config.settings</field>
+    <field name="inherit_id" ref="base_setup.view_general_configuration"/>
+    <field name="arch" type="xml">
+        <xpath expr="//label[@string='Email']/.." position='after'>
+            <group>
+                <label for="partner_names_order" />
+                <div>
+                    <div>
+                        <field name="partner_names_order" class="oe_inline" />
+                        <button name="action_recalculate_partners_name"
+                                string="Recalculate names"
+                                icon="gtk-execute"
+                                type="object"
+                                help="Recalculate names for all partners. This process could take so much time if there are more than 10,000 active partners"/>
```

Comment:
I can run recalculation without warning to user. The problem here is that it is possible that other setting (module_xxx) can change too and this could involve an addon installation process with an action returned. Please see https://github.com/odoo/odoo/blob/8.0/openerp/addons/base/res/res_config.py#L586